thirdparty/libpng/meson.build

```
@@ -0,0 +1,31 @@
+if get_option('builtin_libpng')
+    _libpng_incdirs = include_directories('.')
+
+    _libpng_sources = files([
+        'png.c',
+        'pngerror.c',
+        'pngget.c',
+        'pngmem.c',
+        'pngpread.c',
+        'pngread.c',
+        'pngrio.c',
+        'pngrtran.c',
+        'pngrutil.c',
+        'pngset.c',
+        'pngtrans.c',
+        'pngwio.c',
+        'pngwrite.c',
+        'pngwtran.c',
+        'pngwutil.c',
+    ])
+
+    if PLATFORM == 'android'
+        error('libpng: Need to add in the arm and NEON specific libraries.')
+    endif
```

```suggestion
    if host_machine.cpu_family() == 'aarch64' or host_machine.cpu_family() == 'arm'
        _libpng_sources += files([
            'arm/arm_init.c',
            'arm/filter_neon_intrinsics.c',
            'arm/palette_neon_intrinsics.c',
        ])
    endif
```

This is relevant for iOS, macOS (on Apple Silicon), and ARM Linux not only Android.